Sources/Private/CoreAnimation/CoreAnimationLayer.swift

```diff
@@ -15,28 +15,17 @@ final class CoreAnimationLayer: BaseAnimationLayer {
     animation: Animation,
     imageProvider: AnimationImageProvider,
     fontProvider: AnimationFontProvider,
-    compatibilityTrackerMode: CompatibilityTracker.Mode,
-    didSetUpAnimation: (([CompatibilityIssue]) -> Void)? = nil)
+    compatibilityTrackerMode: CompatibilityTracker.Mode)
+    throws
   {
     self.animation = animation
     self.imageProvider = imageProvider
     self.fontProvider = fontProvider
-    self.didSetUpAnimation = didSetUpAnimation
     compatibilityTracker = CompatibilityTracker(mode: compatibilityTrackerMode)
     super.init()
 
     setup()
-
-    do {
-      try setupChildLayers()
-    } catch {
-      if case CompatibilityTracker.Error.encounteredCompatibilityIssue(let compatibilityIssue) = error {
-        // If the initial layer setup was successful, then the animation will continue
-        // being setup later in `CALayer.display()` below. If this initial setup failed,
-        // then we report the error now.
-        didSetUpAnimation?([compatibilityIssue])
```

Comment:
I reworked this initializer to `throw` after I encountered a silly issue:

```swift
self.animationLayer = CoreAnimationLayer(
  animation: animation,
  compatibilityTrackerMode: .abort,
  didSetUpAnimation: {
     // ...
     makeAnimationLayer(usingEngine: .mainThread)
     // This could run from inside the initializer, meaning the code that takes the result of the initializer
     // and applies it to `self.animationLayer` would run _after_ we finished setting up the main thread layer,
     // putting the `AnimationView` in an inconsistent state.
  })
```